detox/src/android/AndroidExpect.test.js

```diff
@@ -93,6 +93,15 @@ describe('AndroidExpect', () => {
       await e.expect(e.element(e.by.id('test').or(e.by.id('test2')))).toBeVisible();
     });
 
+    it(`matcher immutability`, async () => {
+      const base = e.by.id('test');
+      const modifier = e.by.id('ancestor');
+
+      expect(base.withAncestor(modifier)).not.toBe(base);
+      expect(base.withDescendant(modifier)).not.toBe(base);
+      expect(base.and(modifier)).not.toBe(base);
```

Comment:
More specifically, you can test that `base` and `modifier` haven't mutated, for example:
```js
const base = e.by.id('foo');
const modifier = e.by.id('bar');

const _ = base.withAncestor(modifier);
const __ = base.withDescendant(modifier);
const ___ = base.and(modifier);

expect(base).toEqual(e.by.id('foo'));
expect(modifier).toEqual(e.by.id('bar'));
```

Reply:
also you can separate this to different tests like you did with the `it.each` in `expectTwo.test.js`

Reply:
Thanks, fixed.